File: Golang/Swap Nodes in Pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {

    dummy := &ListNode{Next: head}
    cur := head
    prev := dummy
    keep := &ListNode{Next: nil}
    for cur != nil && cur.Next != nil {
        keep = cur.Next.Next
        cur.Next.Next = cur
        prev.Next = cur.Next
        cur.Next = keep

        prev = cur
        cur = cur.Next
    }

    return dummy.Next
}

File: Python3/Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        dummy.next = head
        cur = head
        prev = dummy
        keep = None
        while cur and cur.next:
            keep = cur.next.next
            cur.next.next = cur
            prev.next = cur.next
            cur.next = keep

            prev = cur
            cur = cur.next

        return dummy.next